aspnetcore/host-and-deploy/aspnet-core-module.md

```diff
@@ -13,12 +13,97 @@ By [Luke Latham](https://github.com/guardrex), [Rick Anderson](https://twitter.c
 
 This document provides instructions on how to configure the ASP.NET Core Module for hosting ASP.NET Core apps. For an introduction to the ASP.NET Core Module and installation instructions, see the [ASP.NET Core Module overview](xref:fundamentals/servers/aspnet-core-module).
 
+::: moniker range=">= aspnetcore-2.2"
+
+## Hosting model
+
+For apps running on .NET Core 2.2 or later, the module supports an in-process hosting model for improved performance compared to reverse-proxy (out-of-process) hosting. For more information, see <xref:fundamentals/servers/aspnet-core-module#aspnet-core-module-description>.
+
+In-procsess hosting is opt-in, but [dotnet new](/dotnet/core/tools/dotnet-new) templates default to the in-process hosting model for all IIS Express scenarios.
+
+To configure an app for in-process hosting, add the `<AspNetCoreModuleHostingModel>` property to the app's project file with a value of `inprocess` (out-of-process hosting is set with `outofprocess`):
+
+```xml
+<PropertyGroup>
+  <AspNetCoreModuleHostingModel>inprocess</AspNetCoreModuleHostingModel>
+</PropertyGroup>
+```
+
+The following characteristics apply when hosting in-process:
+
+* The [Kestrel server](xref:fundamentals/servers/kestrel) isn't used. A custom <xref:Microsoft.AspNetCore.Hosting.Server.IServer> implementation, `IISHttpServer` acts as the app's server.
+
+* The [requestTimeout attribute](#attributes-of-the-aspnetcore-element) doesn't apply to in-process hosting.
+
+* Sharing an app pool among apps isn't supported. Use one app pool per app.
+
+* When using [Web Deploy](/iis/publish/using-web-deploy/introduction-to-web-deploy) or manually placing an [app_offline.htm file in the deployment](xref:host-and-deploy/iis/index#locked-deployment-files), the app might not shut down immediately if there's an open connection. For example, a websocket connection may delay app shut down.
```

Comment:
@pakrym?

Reply:
Is this different from our normal graceful shutdown behavior out of proc?

Reply:
Should be the same.